scene/3d/camera_3d.cpp

```diff
@@ -240,8 +257,44 @@ void Camera3D::set_frustum(real_t p_size, Vector2 p_offset, real_t p_z_near, rea
 	update_gizmos();
 }
 
+void Camera3D::set_custom(Projection p_proj) {
+	if (!force_change && c_proj == p_proj) {
+		return;
+	}
+
+	c_proj = p_proj;
+	mode = PROJECTION_CUSTOM;
+	force_change = false;
+
+	Vector<Plane> planes = c_proj.get_projection_planes(Transform3D());
+	const Projection::Planes intersections[8][3] = {
+		{ Projection::PLANE_FAR, Projection::PLANE_LEFT, Projection::PLANE_TOP },
+		{ Projection::PLANE_FAR, Projection::PLANE_LEFT, Projection::PLANE_BOTTOM },
+		{ Projection::PLANE_FAR, Projection::PLANE_RIGHT, Projection::PLANE_TOP },
+		{ Projection::PLANE_FAR, Projection::PLANE_RIGHT, Projection::PLANE_BOTTOM },
+		{ Projection::PLANE_NEAR, Projection::PLANE_LEFT, Projection::PLANE_TOP },
+		{ Projection::PLANE_NEAR, Projection::PLANE_LEFT, Projection::PLANE_BOTTOM },
+		{ Projection::PLANE_NEAR, Projection::PLANE_RIGHT, Projection::PLANE_TOP },
+		{ Projection::PLANE_NEAR, Projection::PLANE_RIGHT, Projection::PLANE_BOTTOM },
+	};
+
+	for (int i = 0; i < 8; i++) {
+		Plane a = planes[intersections[i][0]];
+		Plane b = planes[intersections[i][1]];
+		Plane c = planes[intersections[i][2]];
+		ERR_FAIL_COND_MSG(!a.intersect_3(b, c), "Camera3D custom projection has non-intersecting planes; skipping update.");
+	}
```

Comment:
My bad! I think this check should be in `set_custom_projection` instead, before `c_proj` is ever set. I'm also not sure if it makes sense to disallow these projections, since it seems like only lighting and the editor gizmo are dependent on having endpoints, if someone familiar could weigh in.

Reply:
Separately, it seems that for testing only the planes for `Transform3D()` is not sufficient, since it is still possible to have no endpoints if the camera transform is different, which means the reported vulkan crash _can_ still happen, albeit rarely (I spent about 30 seconds dragging every projection value back and forth before getting a crash). I don't think testing every time the camera transform changes makes sense, so there should be a more robust way of checking if a projection has endpoints regardless of transform?